cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb

```diff
@@ -20,6 +20,13 @@
   not_if { ::File.exist?(node['cluster']['previous_cluster_config_path']) && !are_queues_updated? }
 end
 
+ruby_block "update_shared_storages" do
+  block do
+    run_context.include_recipe 'aws-parallelcluster-slurm::update_shared_storages'
+  end
+  only_if { are_mount_or_unmount_required? }
```

Comment:
I do not have a strong concern, but I'd like to share with you my thoughts.
I think that `update_head_node_base` should always include `update_shared_storage` because it is a responsibility of the latter to be idempotent and know when it is appropriate to execute and update or not on the storage.
What do you think?

Reply:
I don't understand why we should always execute update_shared_storages?
If mount or unmount are not required, the process doesn't need to take extra time to load the two yaml files

Reply:
My reasoning is as follows, but please consider that I do not have a strong concern, just want to share a design reasoning.

The responsibility of `update_head_node_base` is to know *which are* the components of the system whose state must converge to the one desired by the update. It should not know anything about how such updates are internally executed by each component, but simply the components to involve. In this sense, it knows that there is an update and tells every component to do their job to converge to the desired state and it trust that they do that in an idempotent way.

The responsibility of `update_component_X` (eg update_shared_storage) is to know *what* to do when a change is requested and execute what's needed to converge to the desired state. If the component understands that there is nothing to do, then it returns immediately otherwise it does its job.

Reply:
I got your point. Let me know what you think about it